frontend/src/pages/PipelineDetails.tsx

```diff
@@ -304,21 +304,16 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
     if (templateString) {
       try {
         const template = JsYaml.safeLoad(templateString);
-        if (WorkflowUtils.isArgoWorkflowTemplate(template)) {
```

Comment:
can we have `isTektonPipelineRunTemplate` instead and checking the `kind` and `apiVersion` accordingly? then we can also throw error if the yaml is not pipelinerun crd?

Reply:
Unfortunately, we don't have the Tekton template library in Javascript because the Tekton spec is still dramatically changing on some major releases. We are devoting the spec check to the API which can help us better catch the errors.

Reply:
understood, I checked the code and since the function `isArgoWorkflowTemplate` doesn't enforce schema check. It only check two fields - `kind` and `apiVersion`. these two are common for kube CRD. hence no tekton js lib is needed.
code like this works well:
```
export function isTektonPipelineRunTemplate(template: Workflow): boolean {
  if (template?.kind === 'PipelineRun' && template?.apiVersion?.startsWith('tekton.dev/')) {
    return true;
  }
  return false;
}
```
this could minimize the change in `PipelineDetails.tsx` and the utility function goes to `WorkflowUtils.ts`

similar to my change here: https://github.com/yhwang/kfp-tekton/commit/588ec637ab51d54a98a39e99bc3154acc500a73d

but your change is good too :-)

Reply:
sure, I will make the kind comparison to not be case sensitive to minimize possible future changes.

Reply:
actually I will use your changes since that's the same as Argo implementation.